NebulaPatcher/Patches/Dynamic/UIPlanetDetail_Patch.cs

```diff
@@ -71,11 +68,26 @@ public static bool OnPlanetDataSet_Prefix(UIPlanetDetail __instance)
         [HarmonyPatch(nameof(UIPlanetDetail.RefreshDynamicProperties))]
         public static void OnPlanetDataSet_Postfix(UIPlanetDetail __instance)
         {
-            if(BackupUniverseObserveLevel != -1)
+            if(Multiplayer.IsActive && Multiplayer.Session.IsInLobby)
+            {
+                GameMain.history.universeObserveLevel = GetUniverseObserveLevel();
+            }
+        }
+
+        private static int GetUniverseObserveLevel()
+        {
+            int level = 0;
+            // the tech ids of the 4 tiers of Universe Exploration from https://dsp-wiki.com/Upgrades
+            for (int i = 4104; i >= 4101; i--)
```

Comment:
Another option would be to use `i = 4` and then do `...TechUnlocked(i + 4100))`, then you could just `return i - 1` without needing mod operator.

Reply:
Yeah but then it has to do addition each time while the mod operator only has to get called once.

Ultimately doesn't make much of a difference idk, I liked doing it this way so that it's clear that it's just going through the tech ids 4101-4104 in the loop itself